app/components/provider_interface/safeguarding_declaration_component.rb

```diff
@@ -2,13 +2,43 @@ module ProviderInterface
   class SafeguardingDeclarationComponent < ViewComponent::Base
     include ViewHelper
 
-    attr_reader :message
+    attr_reader :application_choice, :current_provider_user, :details
 
-    def initialize(application_form:)
-      @message = SafeguardingStatus.new(
-        status: application_form.safeguarding_issues_status,
+    def initialize(application_choice:, current_provider_user:)
+      @application_choice = application_choice
+      @current_provider_user = current_provider_user
+    end
+
+    def message
+      set_details if safeguarding_issues_declared? && display_safeguarding_information?
+
+      SafeguardingStatus.new(
+        status: status,
         i18n_key: 'provider_interface.safeguarding_declaration_component',
       ).message
     end
+
+    def display_safeguarding_information?
+      ProviderAuthorisation.new(actor: current_provider_user)
+        .can_view_safeguarding_information?(course: application_choice.course)
+    end
+
+  private
+
+    def status
+      if safeguarding_issues_declared? && !display_safeguarding_information?
+        'has_safeguarding_issues_to_declare_no_permissions'
+      else
+        application_choice.application_form.safeguarding_issues_status
+      end
+    end
+
+    def safeguarding_issues_declared?
+      application_choice.application_form.safeguarding_issues_status == 'has_safeguarding_issues_to_declare'
```

Comment:
`enums` automatically expose helper methods like this:

```suggestion
      application_choice.application_form.has_safeguarding_issues_to_declare?
```

They have the advantage of preventing typos, as it'll raise a `NoMethodError` if you make one.